src/main/java/com/techconative/inmemory/pagination/util/InmemoryFOPS.java

```diff
@@ -0,0 +1,253 @@
+package com.techconative.inmemory.pagination.util;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.techconative.inmemory.pagination.modal.OrderingCriteria;
+import com.techconative.inmemory.pagination.modal.PageResult;
+import com.techconative.inmemory.pagination.modal.PaginationCriteria;
+import exception.InvalidCriteriaException;
+import java.util.*;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Filter, search, sort and paginate your data. PaginationService is the class to be extended to
```

Comment:
@bharathyes check the comments please and update accordingly.

Reply:
Fixed the comments 👍🏾